lib/commands/create-piece.js

```diff
@@ -2,87 +2,83 @@ require('shelljs/global');
+      const stringSet = majorVersion === '2' ? 'a2' : 'a3';
+      const strings = util.getStrings(stringSet, 'create-piece', moduleName);
 
-      const path = 'lib/modules/' + moduleName;
+      util.log('create-piece', `Adding ${moduleName} folder to /${modulesDir}.`);
 
-      mkdir('-p', path);
+      const modulePath = path.join(modulesDir, moduleName);
 
-      const pieceConfig = stripIndent`
-        module.exports = {
-          extend: 'apostrophe-pieces',
-          // TODO: Update the piece type name to be the singular version of the module
-          // name. For example, if the module is "people" the piece type name should be
-          // "person".
-          name: '${pieceName}',
-          // TODO: Update the singular label for editors
-          label: '${util.titleCase(pieceName.replace(/-/g, ' '))}',
-          // TODO: Update the plural label for editors
-          pluralLabel: '${util.titleCase(pieceName.replace(/-/g, ' '))}',
-          addFields: []
-        };
-      `;
+      mkdir('-p', modulePath);
 
-      util.nlog('create-piece', 'Setting up index.js for ' + moduleName + ' module');
-      fs.writeFileSync(path + '/index.js', pieceConfig);
+      util.log('create-piece', `Setting up index.js for ${moduleName} module`);
 
+      const pieceConfig = strings.pieceConfig;
+      fs.writeFileSync(path.join(modulePath, 'index.js'), pieceConfig);
+
+      util.log('create-piece', `YOUR NEXT STEP: add the ${moduleName} module to "modules" in app.js.`);
+
+      // Piece page setup
+      // ****************
       if (options.pages) {
-        util.nlog('create-piece', 'Creating a ' + moduleName + '-pages folder with index.js and appropriate views');
-
-        const pagesConfig = stripIndent`
-          module.exports = {
-            extend: 'apostrophe-pieces-pages',
-            label: '${util.titleCase(pieceName.replace(/-/g, ' '))} Page',
-            addFields: []
-          };
-        `;
-
-        mkdir('-p', path + '-pages');
-        fs.writeFileSync(path + '-pages/index.js', pagesConfig);
-
-        mkdir('-p', path + '-pages/views');
-        exec('touch ' + path + '-pages/views/show.html');
-        exec('touch ' + path + '-pages/views/index.html');
-        util.nlog('create-piece', 'YOUR NEXT STEP: add the ' + moduleName + '-pages module to "modules" in app.js.');
-        util.nlog('create-piece', 'YOUR NEXT STEP: add the ' + pieceName + '-page page type to the "types" array in lib/modules/apostrophe-pages/index.js.');
+        util.log('create-piece', `Creating a ${moduleName}-pages folder with index.js and appropriate views`);
+
+        const pagesConfig = strings.pagesConfig;
+        const pageDir = `${modulePath}-pages`;
+
+        mkdir('-p', path.join(pageDir));
+        fs.writeFileSync(path.join(pageDir, 'index.js'), pagesConfig);
+
+        mkdir('-p', path.join(pageDir, 'views'));
+        exec(`touch ${path.join(pageDir, 'views', 'show.html')}`);
```

Comment:
If you're going to use `path.join` for correctness, you should also be using shell-quote in all of your exec calls that involve user input:

https://github.com/substack/node-shell-quote

Reply:
Or, you could take an approach based on `spawnSync` which would inherently avoid the problem of misquoted or unquoted command line parameter strings with spaces, punctuation, etc.

Reply:
Since you seem to only be doing that for `touch`, you could just use the `touch` npm module.